### PR TITLE
[PWGDQ] Adding dq-Correlations task

### PR DESCRIPTION
**Pull request by victorvalenciatorres**
@mguilbau @sjzhu0v0 @lucamicheletti  First commit of dqCorrelations Task. After discussion with Ionut, to be mainly used for barrel. To be further developped

### CODE REVIEW

**Comment by mguilbau** [comment]
Just to be more precise, this task can be used to analyse pp data. After discussion with Ionut, the strategy for PbPb must be a little different

**Review by lucamicheletti93 — APPROVED** [review]

**Review by iarsene — CHANGES_REQUESTED** [review]
Hi Victor,
In the mattermost channel you mentioned you made a small modification, but here i see a completely new file. Could you clarify?
Also, why are you using both the HistogramManager and HistogramRegistry?
It should be enough to use just the HistogramManager.

File: PWGDQ/Tasks/dqCorrelation.cxx [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+// Declarations of various short names
+using MyEvents = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended>;
+using MyEventsSelected = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended, aod::EventCuts>;
+using MyEventsHashSelected = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended, aod::EventCuts, aod::MixingHashes>;
+using MyEventsVtxCov = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended, aod::ReducedEventsVtxCov>;
+using MyEventsVtxCovSelected = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended, aod::ReducedEventsVtxCov, aod::EventCuts>;
+// using MyEventsVtxCovSelectedQvector = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended, aod::ReducedEventsVtxCov, aod::EventCuts, aod::ReducedEventsQvector>;
+// using MyEventsQvector = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended, aod::ReducedEventsQvector>;
+// using MyEventsHashSelectedQvector = soa::Join<aod::ReducedEvents, aod::ReducedEventsExtended, aod::EventCuts, aod::MixingHashes, aod::ReducedEventsQvector>;
+
```
Comment (iarsene) [inline PWGDQ/Tasks/dqCorrelation.cxx:95]: please remove the commented lines of code

File: PWGDQ/Tasks/dqCorrelation.cxx [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+using MyBarrelTracks = soa::Join<aod::ReducedTracks, aod::ReducedTracksBarrel, aod::ReducedTracksBarrelPID>;
+using MyBarrelTracksWithCov = soa::Join<aod::ReducedTracks, aod::ReducedTracksBarrel, aod::ReducedTracksBarrelCov, aod::ReducedTracksBarrelPID>;
+using MyBarrelTracksSelected = soa::Join<aod::ReducedTracks, aod::ReducedTracksBarrel, aod::ReducedTracksBarrelPID, aod::BarrelTrackCuts>;
+using MyBarrelTracksSelectedWithPrefilter = soa::Join<aod::ReducedTracks, aod::ReducedTracksBarrel, aod::ReducedTracksBarrelPID, aod::BarrelTrackCuts, aod::Prefilter>;
+using MyBarrelTracksSelectedWithCov = soa::Join<aod::ReducedTracks, aod::ReducedTracksBarrel, aod::ReducedTracksBarrelCov, aod::ReducedTracksBarrelPID, aod::BarrelTrackCuts>;
+using MyBarrelTracksSelectedWithColl = soa::Join<aod::ReducedTracks, aod::ReducedTracksBarrel, aod::ReducedTracksBarrelPID, aod::BarrelTrackCuts, aod::ReducedTracksBarrelInfo>;
+// using MyPairCandidatesSelected = soa::Join<aod::Dileptons, aod::DileptonsExtra>;
```
Comment (iarsene) [inline PWGDQ/Tasks/dqCorrelation.cxx:102]: remove

File: PWGDQ/Tasks/dqCorrelation.cxx [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+using MyMuonTracksSelected = soa::Join<aod::ReducedMuons, aod::ReducedMuonsExtra, aod::ReducedMuonsInfo, aod::MuonTrackCuts>;
+using MyMuonTracksWithCov = soa::Join<aod::ReducedMuons, aod::ReducedMuonsExtra, aod::ReducedMuonsInfo, aod::ReducedMuonsCov>;
+using MyMuonTracksSelectedWithCov = soa::Join<aod::ReducedMuons, aod::ReducedMuonsExtra, aod::ReducedMuonsInfo, aod::ReducedMuonsCov, aod::MuonTrackCuts>;
+
+// bit maps used for the Fill functions of the VarManager
+constexpr static uint32_t gkEventFillMap = VarManager::ObjTypes::ReducedEvent | VarManager::ObjTypes::ReducedEventExtended;
+// constexpr static uint32_t gkEventFillMapWithCov = VarManager::ObjTypes::ReducedEvent | VarManager::ObjTypes::ReducedEventExtended | VarManager::ObjTypes::ReducedEventVtxCov;
+//  constexpr static uint32_t gkEventFillMapWithQvector = VarManager::ObjTypes::ReducedEvent | VarManager::ObjTypes::ReducedEventExtended | VarManager::ObjTypes::ReducedEventQvector;
+//  constexpr static uint32_t gkEventFillMapWithCovQvector = VarManager::ObjTypes::ReducedEvent | VarManager::ObjTypes::ReducedEventExtended | VarManager::ObjTypes::ReducedEventVtxCov | VarManager::ObjTypes::ReducedEventQvector;
```
Comment (iarsene) [inline PWGDQ/Tasks/dqCorrelation.cxx:113]: remove

File: PWGDQ/Tasks/dqCorrelation.cxx [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
+constexpr static uint32_t gkTrackFillMap = VarManager::ObjTypes::ReducedTrack | VarManager::ObjTypes::ReducedTrackBarrel | VarManager::ObjTypes::ReducedTrackBarrelPID;
+// constexpr static uint32_t gkTrackFillMapWithCov = VarManager::ObjTypes::ReducedTrack | VarManager::ObjTypes::ReducedTrackBarrel | VarManager::ObjTypes::ReducedTrackBarrelCov | VarManager::ObjTypes::ReducedTrackBarrelPID;
+//  constexpr static uint32_t gkTrackFillMapWithColl = VarManager::ObjTypes::ReducedTrack | VarManager::ObjTypes::ReducedTrackBarrel | VarManager::ObjTypes::ReducedTrackBarrelPID | VarManager::ObjTypes::ReducedTrackCollInfo;
```
Comment (iarsene) [inline PWGDQ/Tasks/dqCorrelation.cxx:116]: remove

File: PWGDQ/Tasks/dqCorrelation.cxx [hunk 5]
```diff
@@ -0,0 +1,512 @@
...
+
+constexpr static uint32_t gkMuonFillMap = VarManager::ObjTypes::ReducedMuon | VarManager::ObjTypes::ReducedMuonExtra;
+//  constexpr static uint32_t gkMuonFillMapWithCov = VarManager::ObjTypes::ReducedMuon | VarManager::ObjTypes::ReducedMuonExtra | VarManager::ObjTypes::ReducedMuonCov;
+
+// constexpr static int pairTypeEE = VarManager::kDecayToEE;
+// constexpr static int pairTypeMuMu = VarManager::kDecayToMuMu;
+// constexpr static int pairTypeEMu = VarManager::kElectronMuon;
```
Comment (iarsene) [inline PWGDQ/Tasks/dqCorrelation.cxx:123]: remove

File: PWGDQ/Tasks/dqCorrelation.cxx [hunk 6]
```diff
@@ -0,0 +1,512 @@
...
+  Configurable<int64_t> nolaterthan{"ccdb-no-later-than", std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count(), "latest acceptable timestamp of creation for the object"};
+  Configurable<std::string> url{"ccdb-url", "http://ccdb-test.cern.ch:8080", "url of the ccdb repository"};
+
+  // // // CCDB access
+  // Service<ccdb::BasicCCDBManager> ccdb;
+  // // CCDB access for the efficiencies and acceptances
+  // Configurable<std::string> fConfigEfficiency{"ccdb-path-efficiency", "Users/r/rcaron/efficiency", "CCDB path to efficiency object"};
+  // Configurable<std::string> fConfigAcceptance{"ccdb-path-acceptance", "", "CCDB path to acceptance or GFWWeights object"};
+  // Configurable<std::string> fConfigURL{"ccdb-url", "http://alice-ccdb.cern.ch", "url of the ccdb repository"};
+  // Configurable<int64_t> fConfigNoLaterThan{"ccdb-no-later-than", std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count(), "latest acceptable timestamp of creation for the object"};
```
Comment (iarsene) [inline PWGDQ/Tasks/dqCorrelation.cxx:174]: remove

File: PWGDQ/Tasks/dqCorrelation.cxx [hunk 7]
```diff
@@ -0,0 +1,512 @@
...
+  ConfigurableAxis axisMultiplicity{"axisMultiplicity", {VARIABLE_WIDTH, 0, 5, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100.1}, "multiplicity / centrality axis for histograms"};
+
+  ////////////////////////////////////////////////////////////////////////////////
+  // Define the filter for events
+  Filter eventFilter = aod::dqanalysisflags::isEventSelected == 1;
+
+  // Define the filter for barrel tracks and forward tracks
+  Filter trackFilter = (nabs(aod::track::eta) <= fConfigCutEtaMax) && (aod::track::pt > fConfigCutPtMin) && (aod::track::pt < fConfigCutPtMax);
+  Filter fwdFilter = (nabs(aod::fwdtrack::eta) <= fConfigMuonCutEtaMax) && (nabs(aod::fwdtrack::eta) >= fConfigMuonCutEtaMin);
+  // Filter fwdFilter = (aod::fwdtrack::eta < -2.f) && (aod::fwdtrack::eta > -4.5f);
```
Comment (iarsene) [inline PWGDQ/Tasks/dqCorrelation.cxx:189]: remove, also see below. Please remove all the lines that are not needed. This helps with readability of the diffs for future PRs

**Review by iarsene — APPROVED** [review]

**Pull request merged**